File: js/src/maplestory/api/dto/characterSetEffectDto.ts
import {CharacterSetEffectInfoDto} from "./characterSetEffectInfoDto";
import {CharacterSetEffectDtoBody} from "../response/characterSetEffectDtoBody";

/**
 * 캐릭터 세트 효과 정보
 */
class CharacterSetEffectDto {

    /**
     * 조회 기준일
     */
    date: Date;

    /**
     * 세트 효과 정보
     */
    setEffect: CharacterSetEffectInfoDto[]

    constructor(obj: CharacterSetEffectDtoBody) {
        const {date, set_effect} = obj;
        this.date = new Date(date);
        this.setEffect = set_effect.map(effect => new CharacterSetEffectInfoDto(effect));
    }
}

export {CharacterSetEffectDto};

File: java/src/main/java/dev/spiralmoon/maplestory/api/dto/character/CharacterSetEffectDTO.java
package dev.spiralmoon.maplestory.api.dto.character;

import com.google.gson.annotations.SerializedName;
import dev.spiralmoon.maplestory.api.Utils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 캐릭터 세트 효과 정보
 */
@AllArgsConstructor
@Data
@ToString
public class CharacterSetEffectDTO {

    /**
     * 조회 기준일 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
     */
    @SerializedName("date")
    private String date;

    /**
     * 세트 효과 정보 목록
     */
    @SerializedName("set_effect")
    private List<CharacterSetEffectInfoDTO> setEffect;

    public LocalDateTime getDate() {
        return date != null
                ? Utils.toLocalDateTime(this.date)
                : null;
    }
}
